rpm/0032-Ignore-Re-Fwd-when-sort-by-subject.patch

```diff
@@ -0,0 +1,77 @@
+index 4af84430..c206e5ff 100644
+--- a/src/libraries/qmfclient/qmailmessagesortkey.h
++++ b/src/libraries/qmfclient/qmailmessagesortkey.h
+@@ -65,7 +65,8 @@ public:
+         ListId,
+         RestoreFolderId,
+         RfcId,
+-        ParentThreadId
++        ParentThreadId,
++        TrimmedSubject
+     };
+ 
+     typedef QMailSortKeyArgument<Property> ArgumentType;
+@@ -96,6 +97,7 @@ public:
+     static QMailMessageSortKey sender(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey recipients(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey subject(Qt::SortOrder order = Qt::AscendingOrder);
++    static QMailMessageSortKey trimmedSubject(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey timeStamp(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey receptionTimeStamp(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey serverUid(Qt::SortOrder order = Qt::AscendingOrder);
+diff --git a/src/libraries/qmfclient/qmailstore_p.cpp b/src/libraries/qmfclient/qmailstore_p.cpp
+index 80c7b955..1252aa9e 100644
+--- a/src/libraries/qmfclient/qmailstore_p.cpp
++++ b/src/libraries/qmfclient/qmailstore_p.cpp
+@@ -627,6 +631,7 @@ static QMap<QMailMessageSortKey::Property, QMailMessageKey::Property> messageSor
+     map.insert(QMailMessageSortKey::Sender, QMailMessageKey::Sender);
+     map.insert(QMailMessageSortKey::Recipients, QMailMessageKey::Recipients);
+     map.insert(QMailMessageSortKey::Subject, QMailMessageKey::Subject);
++    map.insert(QMailMessageSortKey::TrimmedSubject, QMailMessageKey::Subject);
+     map.insert(QMailMessageSortKey::TimeStamp, QMailMessageKey::TimeStamp);
+     map.insert(QMailMessageSortKey::ReceptionTimeStamp, QMailMessageKey::ReceptionTimeStamp);
+     map.insert(QMailMessageSortKey::Status, QMailMessageKey::Status);
+@@ -1704,7 +1709,20 @@ QString buildOrderClause(const ArgumentListType &list, const QString &alias)
+             field = QString::fromLatin1("(%1 & %2)").arg(field).arg(QString::number(arg.mask));
+         }
+         if (caseInsensitiveProperty(arg.property)) {
+-            sortColumns.append(QLatin1String("ltrim(") + field + QLatin1String(",'\\\"') COLLATE NOCASE ") +
++            if (arg.property == QMailMessageSortKey::TrimmedSubject) {
++                static const QList<QLatin1String> prefixes = {
++                    QLatin1String("Re:"),
++                    QLatin1String("RE:"),
++                    QLatin1String("Fw:"),
++                    QLatin1String("FW:"),
++                    QLatin1String("Fwd:"),
++                    QLatin1String("FWD:")
++                };
++                static const QString trimTemplate = QStringLiteral("replace(%1, '%2', '')");
++                for (const QLatin1String &prefix : prefixes)
++                    field = trimTemplate.arg(field, prefix);
```

Comment:
I may be misreading this, but am I correct in thinking that this will remove all instances of "Re:" etc. from the string before sorting, independent of the position? So for example "HERE: a message" will be sorted as if it were "HE a message"?

Is there any way to have it only remove prefixed strings?

Reply:
Fixed.